File: backend/job-application-service/src/main/java/com/jobportal/jobapp/dto/JobRequestDto.java
package com.jobportal.jobapp.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class JobRequestDto {

    @NotNull(message = "Employer ID is required")
    private Integer empId;

    @NotBlank(message = "Job title is required")
    private String title;

    @NotBlank(message = "Job description is required")
    private String description;

    @NotBlank(message = "Role name is required")
    private String role;

    private BigDecimal experience;

    private BigDecimal salary;

    @NotBlank(message = "Location is required")
    private String location;

    private Integer state;

    private Integer city;

    @NotBlank(message = "Job type is required (e.g. FT, PT, Intern, Contract, Remote)")
    private String type;

    private String status; // 'Open', 'Closed', 'Draft'

    public JobRequestDto() {
    }

    public Integer getEmpId() {
        return empId;
    }

    public void setEmpId(Integer empId) {
        this.empId = empId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public BigDecimal getExperience() {
        return experience;
    }

    public void setExperience(BigDecimal experience) {
        this.experience = experience;
    }

    public BigDecimal getSalary() {
        return salary;
    }

    public void setSalary(BigDecimal salary) {
        this.salary = salary;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Integer getState() {
        return state;
    }

    public void setState(Integer state) {
        this.state = state;
    }

    public Integer getCity() {
        return city;
    }

    public void setCity(Integer city) {
        this.city = city;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: backend-dotnet/Services/JobPortal.JobApplicationService/DTOs/JobRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace JobPortal.JobApplications.DTOs;

public class JobRequestDto
{
    [Required(ErrorMessage = "Employer ID is required")]
    public int? EmpId { get; set; }

    [Required(AllowEmptyStrings = false, ErrorMessage = "Job title is required")]
    public string Title { get; set; } = string.Empty;

    [Required(AllowEmptyStrings = false, ErrorMessage = "Job description is required")]
    public string Description { get; set; } = string.Empty;

    [Required(AllowEmptyStrings = false, ErrorMessage = "Role name is required")]
    public string Role { get; set; } = string.Empty;

    public decimal? Experience { get; set; }

    public decimal? Salary { get; set; }

    [Required(AllowEmptyStrings = false, ErrorMessage = "Location is required")]
    public string Location { get; set; } = string.Empty;

    public int? State { get; set; }

    public int? City { get; set; }

    [Required(AllowEmptyStrings = false, ErrorMessage = "Job type is required (e.g. FT, PT, Intern, Contract, Remote)")]
    public string Type { get; set; } = string.Empty;

    public string? Status { get; set; }
}
